docs(llmobs): document MultiEvaluatorResult for experiment evaluators (#37019)
DataDog/documentation: content/en/llm_observability/evaluations/evaluation_developer_guide.md

- **[EvaluatorContext](#evaluatorcontext)**: The input to an evaluator. Contains the LLM's input, output, expected output, and span identifiers. In Experiments, the SDK builds this automatically from each dataset record. In production, you construct the EvaluatorContext yourself.
- **[EvaluatorResult](#evaluatorresult)**: The output of an evaluator. Contains a typed value, optional reasoning, a pass/fail assessment, metadata, and tags. You can also return a plain value (`str`, `float`, `int`, `bool`, `dict`) instead.

> - **[MultiEvaluatorResult](#multievaluatorresult)**: An optional container for returning **multiple named metrics** from a single evaluator. Each sub-value can be a plain value or its own `EvaluatorResult`. Useful when one evaluator pass produces several related metrics (for example, precision, recall, and F1).

Reviewer: None of the other bullets in this list use inline bold for emphasis; suggest dropping it for consistency.

```suggestion
- **[MultiEvaluatorResult](#multievaluatorresult)**: An optional container for returning multiple named metrics from a single evaluator. Each sub-value can be a plain value or its own `EvaluatorResult`. Useful when one evaluator pass produces several related metrics (for example, precision, recall, and F1).
```

    input_data: Any,
    output_data: Any,
    expected_output: Any
) -> Union[JSONType, EvaluatorResult, MultiEvaluatorResult]:
    ...
{{< /code-block >}}

> You can return either:

Reviewer: "Either" implies exactly two choices, but the list now has three items. The PR added a third option without updating the lead-in.

```suggestion
You can return:
```

def confusion_matrix_evaluator(input_data, output_data, expected_output):
    predicted = bool(output_data)
    expected = bool(expected_output)
    correct = predicted == expected

    if predicted and expected:
        category = "true_positive"
    elif predicted and not expected:
        category = "false_positive"
    elif not predicted and expected:
        category = "false_negative"
    else:
        category = "true_negative"

    return MultiEvaluatorResult(
        {
            "correct": EvaluatorResult(
                value=correct,
                assessment="pass" if correct else "fail",
                reasoning=f"predicted={predicted}, expected={expected}",
            ),
            "category": category,
            "false_positive": category == "false_positive",
        }
    )
{{< /code-block >}}

By default, sub-metric labels are prefixed with the evaluator's name: `confusion_matrix_evaluator-correct`, `confusion_matrix_evaluator-category`, and so on. To emit raw keys without a prefix, pass `prefix=False`:

{{< code-block lang="python" >}}
return MultiEvaluatorResult({"precision": 0.9, "recall": 0.8}, prefix=False)
{{< /code-block >}}

> <div class="alert alert-warning">If two evaluators in the same row emit the same metric label (for example, both use <code>prefix=False</code> with the same key), the second value overwrites the first and a warning is logged.</div>

Reviewer: "Row" isn't used elsewhere on this page. The established term is "record" or "dataset record."

```suggestion
<div class="alert alert-warning">If two evaluators emit the same metric label for the same dataset record (for example, both use <code>prefix=False</code> with the same key), the second value overwrites the first and a warning is logged.</div>
```

`MultiEvaluatorResult` is also supported in class-based evaluators and summary evaluators:

{{< code-block lang="python" >}}
from ddtrace.llmobs import BaseEvaluator, EvaluatorContext, MultiEvaluatorResult

class ConfusionMatrixEvaluator(BaseEvaluator):
    def __init__(self):
        super().__init__(name="confusion_matrix")

    def evaluate(self, context: EvaluatorContext) -> MultiEvaluatorResult:
        predicted = bool(context.output_data)
        expected = bool(context.expected_output)
        correct = predicted == expected
        return MultiEvaluatorResult(
            {
                "correct": correct,
                "false_positive": predicted and not expected,
                "false_negative": not predicted and expected,
            },
            prefix=False
        )

>         # Emitted labels: confusion_matrix-correct, confusion_matrix-false_positive, ...

Reviewer: This comment falls after the `return` statement, making it unreachable code. Consider moving it to just before the `return`, or folding the emitted label names into the surrounding prose.

            "category": category,
            "false_positive": category == "false_positive",
        }
    )
{{< /code-block >}}

By default, sub-metric labels are prefixed with the evaluator's name: `confusion_matrix_evaluator-correct`, `confusion_matrix_evaluator-category`, and so on. To emit raw keys without a prefix, pass `prefix=False`:

{{< code-block lang="python" >}}
return MultiEvaluatorResult({"precision": 0.9, "recall": 0.8}, prefix=False)
{{< /code-block >}}

#### `MultiEvaluatorResult` parameters

| Parameter | Type | Default | Description |
|-----------|------|---------|-------------|
| `values` | `dict[str, Union[str, float, int, bool, dict, EvaluatorResult]]` | — | Mapping of sub-metric name to value or `EvaluatorResult`. Keys must follow evaluation [label naming conventions](#naming-conventions). |
| `prefix` | `bool` | `True` | If `True`, labels are emitted as `"<evaluator_name>-<key>"`. If `False`, labels are the raw keys. |

<div class="alert alert-warning">If two evaluators in the same row emit the same metric label (for example, both use <code>prefix=False</code> with the same key), the second value overwrites the first and a warning is logged.</div>

> `MultiEvaluatorResult` is also supported in class-based evaluators and summary evaluators:

Reviewer 2: have you tested this too?

Author: I did and it works well.

| `metadata` | `Optional[Dict[str, Any]]` | Additional metadata about the evaluation. |
| `tags` | `Optional[Dict[str, str]]` | Tags to apply to the evaluation metric. |

### MultiEvaluatorResult

A container for emitting multiple named evaluation metrics from a single evaluator. Import from `ddtrace.llmobs`.

| Parameter | Type | Default | Description |
|-----------|------|---------|-------------|

> | `values` | `dict[str, Union[str, float, int, bool, dict, EvaluatorResult]]` | — | Mapping of sub-metric name to a plain value or an `EvaluatorResult`. Must be non-empty; keys must follow evaluation [label naming conventions](#naming-conventions). |

Reviewer: Every other type in the data model reference tables uses `typing`-style capitalized generics (`Dict`, `List`, `Optional`). This entry is the only one using lowercase PEP 585 syntax. Please capitalize to match.

```suggestion
| `values` | `Dict[str, Union[str, float, int, bool, dict, EvaluatorResult]]` | — | Mapping of sub-metric name to a plain value or an `EvaluatorResult`. Must be non-empty; keys must follow evaluation [label naming conventions](#naming-conventions). |
```

| `metadata` | `Optional[Dict[str, Any]]` | Additional metadata about the evaluation. |
| `tags` | `Optional[Dict[str, str]]` | Tags to apply to the evaluation metric. |

> ### MultiEvaluatorResult

Reviewer 2: this is very similar to the "#### `MultiEvaluatorResult` parameters" section do we need both?

Author: I updated the doc to remove the table which appears twice
